src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -388,12 +426,21 @@ void IList.Remove(object value)
 
         void ICollection.CopyTo(Array array, int index)
         {
-            if (array == null)
+            if ((array != null) && (array.Rank != 1))
+            {
                 throw new ArgumentNullException(nameof(array));
-            if (array.Rank != 1)
-                throw new ArgumentException(nameof(array));
+            }
+            Contract.EndContractBlock();
 
-            Array.Copy(_items, 0, array, index, _size);
+            try
+            {
+                // Array.Copy will check for NULL.
```

Comment:
It will, but you already have to branch on `array != null` to avoid throwing when you test for `array.Rank != 1`, so while this could save a test and branch otherwise, as it is you have to include both and it would be a clearer exception behaviour IMO to throw ANE at that point.